packages/integration-tests/src/EntityManager.clone.test.ts

```diff
@@ -0,0 +1,270 @@
+    expect(a2.changes.publisher.originalValue).toBe(undefined);
+  });
+
+  it("can clone polymorphic references", async () => {
+    const em = newEntityManager();
+    // Given an entity that is a polymorphic parent of two children
+    const a1 = newAuthor(em, { comments: [{}, {}] });
+    await em.flush();
+    // When we clone the entity
+    const a2 = await em.clone(a1, { hint: "comments" });
+    await em.flush();
+    // Then we expect the cloned entity to have cloned copies of all its nested references
+    expect(a2.comments.get.length).toBe(2);
+    expect(a2.comments.get[0].id).toBe("comment:3");
+    expect(a2.comments.get[1].id).toBe("comment:4");
+    expect(a2.comments.get[0].parent.get).toBe(a2);
+  });
+
+  it("can clone a collection of entities", async () => {
+    const em = newEntityManager();
+
+    // Given an author with 2 books, both of which have comments
+    const a1 = newAuthor(em, {
+      firstName: "a1",
+      books: [
+        { title: "b1", comments: [{ text: "Great book!" }] },
+        { title: "b2", comments: [{ text: "Ok book" }] },
+      ],
+    });
+    await em.flush();
+
+    // When I ask to clone just the books
+    const [b3, b4] = await em.clone(a1.books.get, {
+      hint: "comments",
+    });
+    await em.flush();
+
+    // Then we expect the author to now have 4 books
+    expect(a1.books.get).toHaveLength(4);
+    // and b3/b4 have the same titles as b1/b2, but be different entities
+    const [b1, b2] = a1.books.get;
+    expect(b3.title).toEqual(b1.title);
+    expect(b3).not.toEqual(b1);
+    expect(b4.title).toEqual(b2.title);
+    expect(b4).not.toEqual(b2);
+    // And only one author exists, and comments were also cloned
+    expect(await numberOf(em, Author, Book, Comment)).toEqual([1, 4, 4]);
+  });
+
+  it("calls postClone for each cloned entity", async () => {
+    const em = newEntityManager();
+
+    // Given an author with 2 books, both of which have comments
+    const a1 = newAuthor(em, {
+      firstName: "a1",
+      books: [
+        { title: "b1", comments: [{ text: "Great book!" }] },
+        { title: "b2", comments: [{ text: "Ok book" }] },
+      ],
+    });
+    await em.flush();
+    // When we clone the author
+    const postClone = jest.fn();
+    await em.clone(a1, { hint: { books: "comments" }, postClone });
+    // Then I expect `postClone` was called once for each entity to be cloned
+    expect(postClone).toHaveBeenCalledTimes(5);
+  });
+
+  describe("skipIf", () => {
+    it("can skip the main entity you asked to clone", async () => {
+      const em = newEntityManager();
+      // Given an entity to clone
+      const p1 = newPublisher(em, { name: "p1" });
+      await em.flush();
+      // When I attempt to clone it, but ask to skip the entity
+      await expect(em.clone(p1, { skipIf: (e) => e === p1 }))
+        // Then I expect it to fail
+        .rejects.toThrow("no entities were cloned given the provided option");
```

Comment:
though: This seemed like the `best` thing to do in this case, but open to suggestions